roles/idrac_gather_facts/tests/asserts/controller_assert.yml

```diff
@@ -10,30 +11,67 @@
     body_format: "{{ idrac_gather_facts_uri_body_format }}"
     status_code: "{{ idrac_gather_facts_uri_status_code }}"
     return_content: "{{ idrac_gather_facts_uri_return_content }}"
-  register: controller_result
+  register: storage_entity
   no_log: true
 
-- name: Set controller facts
+- name: Get storage entities
   ansible.builtin.set_fact:
-    api_response: "{{ controller_result.json | ansible.utils.remove_keys(target=['@odata.context', '@odata.id', '@odata.type']) }}"
+    collected_storage_entity: "{{ (collected_storage_entity | default([])) + [item['@odata.id']] }}"
+  loop: "{{ storage_entity.json.Members }}"
 
-- name: Set the keys diff
+- name: Get Controllers URL information.
+  ansible.builtin.uri:
```

Comment:
can't we have same module_defaults here?

Reply:
Used referenced variables instead.